crates/chain/src/tx_data_traits.rs

```diff
@@ -100,24 +100,42 @@ impl Anchor for ConfirmationBlockTime {
     }
 }
 
-/// An [`Anchor`] that can be constructed from a given block, block height and transaction position
-/// within the block.
-pub trait AnchorFromBlockPosition: Anchor {
-    /// Construct the anchor from a given `block`, block height and `tx_pos` within the block.
-    fn from_block_position(block: &bitcoin::Block, block_id: BlockId, tx_pos: usize) -> Self;
+/// Minimal set of parameters that are needed to construct a generic [`Anchor`].
+/// Typically used as an additional constraint on anchor:
+/// `A: Anchor + From<BlockPosition<'b>>`.
+pub struct BlockPosition<'b> {
```

Comment:
I think this should be called `BlockAndHeight` and should be documented simply as that. I'm not sure it should be described as the "minimal" set of parameters. It's more like like the maximal amount of detail about the block that could be needed to make an anchor i.e. any anchor should be able to be made from a `BlockPosition`. We could even type constraint that `Anchor: for<'a> From<BlockPosition<'a>>`.

Reply:
If we were to have a merkle-proof anchor (which I think we should have), don't we also need a tx position in block? I'm not too great with naming, but I'm thinking `TxPosInBlock`?

```rust
pub struct TxPosInBlock<'b> {
    block: &'b Block,
    height: u32,
    tx_pos: usize, 
}

Reply:
Also, we are missing some derives. `#[derive(Debug, Clone, Copy, PartialEq, Eq)]` I think makes sense for this type.

Reply:
> Also, we are missing some derives. #[derive(Debug, Clone, Copy, PartialEq, Eq)] I think makes sense for this type.

Added, thanks.

Reply:
All fields should be public as well, otherwise external modules/crates can't impl `From<BlockPosition>`.

Reply:
Hmm, it is more of my habit to start with hidden fields (in order to hide internals) and use getters. The blockhash was a bonus.

However, as BDK typically exposes publicly all fields, I guess it would be appropriate to follow also in this case.

Reply:
> We could even type constraint that `Anchor: for<'a> From<BlockPosition<'a>>`.

`From<BlockPosition>` is currently used only by two methods. Wouldn't that be too strict to require that of every future implementation of Anchor?

> I'm not sure it should be described as the "minimal" set of parameters. It's more like like the maximal amount of detail about the block that could be needed to make an anchor 

Mmm, yeah, I see how it can be unclear. My idea was “if you have _at least_ these parameters at disposal, you can create any anchor.“ I will reword.

Reply:
Yes @evanlinjin's design looks right to me.

> From<BlockPosition> is currently used only by two methods. Wouldn't that be too strict to require that of every future implementation of Anchor?

I think if you can't get your anchor from the full block and position you may be using anchors in a way you probably shouldn't.

Reply:
> If we were to have a merkle-proof anchor (which I think we should have), don't we also need a tx position in block?

If we decide this new struct should include the position of the tx in block, then the original name `BlockPosition` would again makes sense to me but I won't belabor the point.

Reply:
`BlockPosition` may be interpreted as position of block in the Blockchain though